tests/unit/commands/test_lint.py

```diff
@@ -138,114 +209,719 @@ def test_lint_http_https_proxy(
+    with snap_file.open("w") as yaml_file:
+        print(
+            dedent(
+                """\
+                name: test-name
+                version: "1.0"
+                summary: test summary
+                description: test description
+                base: core22
+                confinement: strict
+                grade: stable
+                architectures: [amd64, arm64, armhf]
+
+                apps:
+                  app1:
+                    command: app1
+                    command-chain: [fake-command]
+                    extensions: [fake-extension]
+
+                parts:
+                  nil:
+                    plugin: nil
+                    parse-info:
+                      - usr/share/metainfo/app1.appdata.xml
+                    stage-packages:
+                      - mesa-opencl-icd
+                      - ocl-icd-libopencl1
+                      - on amd64:
+                        - intel-opencl-icd
+                """
+            ),
+            file=yaml_file,
+        )
+
+    result = LintCommand(None)._load_project(snapcraft_yaml_file=snap_file)
+    assert result == Project.unmarshal(
+        {
+            "name": "test-name",
+            "base": "core22",
+            "build_base": "core22",
+            "version": "1.0",
+            "summary": "test summary",
+            "description": "test description",
+            "confinement": "strict",
+            "grade": "stable",
+            "architectures": [{"build_on": ["amd64"], "build_for": ["amd64"]}],
+            "apps": {
+                "app1": {
+                    "command": "app1",
+                    "plugs": ["fake-plug"],
+                    "command-chain": ["fake-command"],
+                }
+            },
+            "parts": {
+                "nil": {
+                    "plugin": "nil",
+                    "stage-packages": [
+                        "mesa-opencl-icd",
+                        "ocl-icd-libopencl1",
+                        "intel-opencl-icd",
+                    ],
+                    "after": ["fake-extension/fake-part"],
+                },
+                "fake-extension/fake-part": {"plugin": "nil"},
+            },
+        }
+    )
+
+
+def test_load_project_no_file(emitter, tmp_path):
+    """Return None if there is no snapcraft.yaml file."""
+    snapcraft_yaml_file = tmp_path / "snap/snapcraft.yaml"
+
+    result = LintCommand(None)._load_project(snapcraft_yaml_file=snapcraft_yaml_file)
+
+    assert not result
+    emitter.assert_debug(f"Could not find {snapcraft_yaml_file.name!r}.")
+
+
+@pytest.mark.parametrize("base", ["core", "core18", "core20"])
+def test_load_project_unsupported_core_error(base, tmp_path):
+    """Raise an error if for snaps with core, core18, and core20 bases."""
+    # create a simple snapcraft.yaml
+    (tmp_path / "snap").mkdir()
+    snap_file = tmp_path / "snap/snapcraft.yaml"
+    with snap_file.open("w") as yaml_file:
```

Comment:
it's a (very) minor suggestion but `yaml_file.write_text(dedent(...))` might look cleaner

Reply:
Good suggestion!